File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
import java.util.stream.Collectors;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;

        Deque<Integer> q = new ArrayDeque<>();
        Deque<Integer> idx = new ArrayDeque<>();

        List<Integer> lst = Arrays.stream(priorities)
                .boxed()
                .sorted()
                .collect(Collectors.toList());

        for(int i=0; i<priorities.length; i++){
            q.add(priorities[i]);
            idx.add(i);
        }

        while(lst.size()>0){
            int x = q.removeFirst();
            int y = idx.removeFirst();
            if(x == lst.get(lst.size()-1)){
                lst.remove(lst.size() - 1);
                if(y==location){
                    return answer;
                }
                answer += 1;
            }else{
                q.add(x);
                idx.add(y);
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    q = deque()
    idx = deque()
    l = sorted(priorities)
    
    for i, j in enumerate(priorities):
        q.append(j)
        idx.append(i)
    answer = 1
    while len(q) > 0:
        x, y = q.popleft(), idx.popleft()
        if x == l[-1]:
            l.pop()
            if y == location:
                return answer
            answer += 1
        else:
            q.append(x)
            idx.append(y)